### PR TITLE
[UTOPIA-1044]add more unit test for ppq and auth module and comments

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

-
Screenshot 2023-04-18 at 1 04 59 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class AuthController {
 
   @Get('user')
   async getUserInfo(@Req() req) {
-    if (req.headers.authorization === 'Bearer') return;
+    if (!req.headers.authorization.startsWith('Bearer')) return;
```
Comment (binggaofw) [inline src/backend/src/modules/auth/auth.controller.ts:48]: i just realize this bug when create test, here should check the authorization start with Bearer instead ===, please le t me know if you have any concerns

Comment (kushal-arora-fw) [reply src/backend/src/modules/auth/auth.controller.ts:48]: Out apis are protected by auth guards. User can never reach this statement without a Bearer token.

File: src/backend/test/unit/auth/auth.controller.spec.ts [hunk 2]
```diff
@@ -55,14 +94,61 @@ describe('AuthController', () => {
...
   describe('getUserInfo', () => {
+    /**
+     * @Description
+     * This test validates the non happy flow if the method `controller.getUserInfo` is called with incorrect mock data
+     *
+     * @Input
+     *   - accessToken
+     *   - request
+     *
+     * @Output 201
```
Comment (kushal-arora-fw) [inline src/backend/test/unit/auth/auth.controller.spec.ts:124]: is it 201?

File: src/backend/test/unit/auth/auth.controller.spec.ts [hunk 3]
```diff
@@ -55,14 +94,61 @@ describe('AuthController', () => {
...
+      expect(service.getUserInfo).not.toHaveBeenCalled();
+    });
+    /**
+     * @Description
+     * This test validates the happy flow if the method `authService.getUserInfo` is called with correct mock data
+     *
+     * @Input
+     *   -Request
+     *
+     * @Output 201
```
Comment (kushal-arora-fw) [inline src/backend/test/unit/auth/auth.controller.spec.ts:141]: will this return 201? I don't think so

**Review by kushal-arora-fw — COMMENTED** [review]
I think you need to update a lot of comments. Many of them say output 201. which will not be the case.

File: src/backend/src/modules/auth/auth.controller.ts (hunk 1 above)
Comment (kushal-arora-fw) [reply src/backend/src/modules/auth/auth.controller.ts:48]: but sure, we can keep it. instead can you please change to `Bearer ` 
Bearer with a space?

File: src/backend/test/unit/auth/auth.controller.spec.ts (hunk 2 above)
Comment (kushal-arora-fw) [reply src/backend/test/unit/auth/auth.controller.spec.ts:124]: usually they are when we create a new resource.

File: src/backend/test/unit/auth/auth.controller.spec.ts [hunk 4]
```diff
@@ -82,7 +82,7 @@ describe('AuthController', () => {
      * @Input
      *   - Authorization code
      *
-     * @Output 201
+     * @Output 200
```
Comment (Bottle7) [inline src/backend/test/unit/auth/auth.controller.spec.ts:85]: I'm pretty sure that this is actually a 201 due to the fact that it creates something. The swagger docs actually show the 'callback' as returning 201 (undocumented) on use.

Comment (binggaofw) [reply src/backend/test/unit/auth/auth.controller.spec.ts:85]: good catch, i will change to 201

**Review by Bottle7 — APPROVED** [review]
🚀

**Review by kushal-arora-fw — APPROVED** [review]
All good except a tiny change

**Pull request merged**